front/components/data_source/MultipleDocumentsUpload.tsx

```diff
@@ -32,69 +33,161 @@ export const MultipleDocumentsUpload = ({
 }: MultipleDocumentsUploadProps) => {
   const fileInputRef = useRef<HTMLInputElement>(null);
   const [isLimitPopupOpen, setIsLimitPopupOpen] = useState(false);
-  const [clicked, setClicked] = useState(false);
-  useEffect(() => {
-    if (isOpen && !clicked) {
-      fileInputRef.current?.click();
-      setClicked(true);
-    }
-  }, [isOpen, onClose, clicked]);
+  const [wasOpened, setWasOpened] = useState(isOpen);
 
   const [isBulkFilesUploading, setIsBulkFilesUploading] = useState<null | {
     total: number;
     completed: number;
   }>(null);
 
+  const close = useCallback(
+    (save: boolean) => {
+      // Clear the values of the file input
+      if (fileInputRef.current) {
+        fileInputRef.current.value = "";
+      }
+      onClose(save);
+    },
+    [onClose]
+  );
+
   const sendNotification = useSendNotification();
 
-  const handleUpsert = async (text: string, documentId: string) => {
-    const body: PostDataSourceWithNameDocumentRequestBody = {
-      name: documentId,
-      timestamp: null,
-      parents: null,
-      section: {
-        prefix: null,
-        content: text,
-        sections: [],
-      },
-      text: null,
-      source_url: undefined,
-      tags: [],
-      light_document_output: true,
-      upsert_context: null,
-      async: false,
-    };
+  const handleUpsert = useCallback(
+    async (text: string, documentId: string) => {
+      const body: PostDataSourceWithNameDocumentRequestBody = {
+        name: documentId,
+        timestamp: null,
+        parents: null,
+        section: {
+          prefix: null,
+          content: text,
+          sections: [],
+        },
+        text: null,
+        source_url: undefined,
+        tags: [],
+        light_document_output: true,
+        upsert_context: null,
+        async: false,
+      };
+
+      try {
+        const res = await fetch(
```

Comment:
Why not moving this to the "swr" folder?

Reply:
out of scope (and I have more urgent matters) but i'll do it next time i have the opportunity.